packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2928,7 +2928,28 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       widget.controller.addListener(_didChangeTextEditingValue);
       _updateRemoteEditingValueIfNeeded();
     }
-    if (widget.controller.selection != oldWidget.controller.selection) {
+
+    if (_selectionOverlay != null
+        && (widget.contextMenuBuilder != oldWidget.contextMenuBuilder
+            || widget.selectionControls != oldWidget.selectionControls
+            || widget.onSelectionHandleTapped != oldWidget.onSelectionHandleTapped
+            || widget.dragStartBehavior != oldWidget.dragStartBehavior
+            || widget.magnifierConfiguration != oldWidget.magnifierConfiguration)) {
+      final bool shouldShowToolbar = _selectionOverlay!.toolbarIsVisible;
+      final bool shouldShowHandles = _selectionOverlay!.handlesVisible;
+      _selectionOverlay!.dispose();
```

Comment:
is the dispose necessary? it's omitted on line 4290.

Reply:
I played around with this and yes it's necessary, things from the old overlay stick around on the screen if it's not there. It looks like 4290 should have made that assignment only when it was null, which I'll fix.